services/pull/update.go

```diff
@@ -104,11 +104,25 @@ func IsUserAllowedToUpdate(pull *models.PullRequest, user *user_model.User) (mer
 		return false, false, nil
 	}
 
+	baseRepoPerm, err := models.GetUserRepoPermission(pull.BaseRepo, user)
```

Comment:
niche that's the reason I original created this func :)

relay happy to see it coming to live